p3g/client.py

```diff
@@ -1,23 +1,31 @@
+import argparse
 import subprocess
 import sys
+from importlib.metadata import version
 
 
 def main():
-    if len(sys.argv) < 2:
-        print("Usage: p3g <command>")
-        print("Available commands: generate")
-        sys.exit(1)
-
-    command = sys.argv[1]
-
-    if command == "generate":
-        generate_proj()
+    parser = argparse.ArgumentParser(description="P3G CLI tool")
+    subparsers = parser.add_subparsers(dest="command")
+    generate_parser = subparsers.add_parser("generate", help="Generate a new project")
+    generate_parser.set_defaults(func=generate_proj)
+    parser.add_argument(
+        "--version", action="version", version=f'%(prog)s {version("p3g")}'
+    )
+    args = parser.parse_args()
+
+    if hasattr(args, "func"):
+        args.func()
     else:
-        print(f"Unknown command: {command}")
+        print(f"Unknown command: {args.command}")
+        parser.print_help()
+        sys.exit(1)
 
 
 def generate_proj():
-    subprocess.run(["cookiecutter", "gh:Undertone0809/P3G", "--checkout", "v1.3.1"])
+    subprocess.run(
+        ["cookiecutter", "gh:Undertone0809/P3G", "--checkout", version("p3g")]
```

Comment:
@Undertone0809 I think this line broke the installation logic. The return value of `version("p3g")` is something like `1.3.4`... where the tags are prepended with `v` i.e. `v1.3.4`.

When I invoke `p3g generate`, I get an error
```
The 1.3.4 branch of repository https://github.com/Undertone0809/P3G.git could not found, have you made a typo?
```

I will submit a PR for this now.